### PR TITLE
Node.js bindings: Add JS bindings for SPI

### PR DESCRIPTION
**Pull request by nagineni**
This patch contains the native and highlevel JavaScript bindings for SPI.

### CODE REVIEW

**Comment by nagineni** [comment]
@gabrielschulhof @barbieri could you please review this?

File: bindings/nodejs/src/functions/spi.cc [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        return;
+    }
+    memset(rxBuffer, 0, length * sizeof(uint8_t));
+    Nan::Callback *callback =
+        new Nan::Callback(Local<Function>::Cast(info[2]));
+    bool returnValue =
+        sol_spi_transfer(spi, txBuffer, rxBuffer, length,
+            sol_spi_transfer_cb, callback);
+
+    if (!returnValue || !hijack_ref()) {
```
Comment (gabrielschulhof) [inline bindings/nodejs/src/functions/spi.cc]: You should make sure that hijacking has not failed before attempting to transfer the data.

Comment (gabrielschulhof) [inline bindings/nodejs/src/functions/spi.cc]: ... and you should also not set the return value if it does fail.

Comment (nagineni) [inline bindings/nodejs/src/functions/spi.cc]: Thanks for reviewing @gabrielschulhof . I will create a new PR with the fix.

**Pull request closed without merging**